File: src/main/java/me/ag2s/epublib/domain/ResourceReference.java
package me.ag2s.epublib.domain;

import java.io.Serializable;

public class ResourceReference implements Serializable {
   private static final long serialVersionUID = 2596967243557743048L;
   protected Resource resource;

   public ResourceReference(Resource resource) {
      this.resource = resource;
   }

   public Resource getResource() {
      return this.resource;
   }

   public void setResource(Resource resource) {
      this.resource = resource;
   }

   public String getResourceId() {
      return this.resource != null ? this.resource.getId() : null;
   }
}

File: src/main/kotlin/me/ag2s/epublib/domain/ResourceReference.kt
package me.ag2s.epublib.domain

import java.io.Serializable

open class ResourceReference(@JvmField protected var resource: Resource?) : Serializable {
    open fun getResource(): Resource? = resource
    open fun setResource(resource: Resource?) { this.resource = resource }
    fun getResourceId(): String? = resource?.getId()
    companion object { private const val serialVersionUID = 2596967243557743048L }
}
